test/controllers/FacetQueryControllerTest.ts

```diff
@@ -197,6 +199,212 @@ export function FacetQueryControllerTest() {
           })
         );
       });
+
+      describe('with allowed values options on the facet', () => {
+        let facetSearchParams: FacetSearchParameters;
+        let searchPromise: Promise<IQueryResults>;
+
+        beforeEach(() => {
+          searchPromise = new Promise((resolve, reject) => {
+            const results = FakeResults.createFakeResults();
+            const singleGroupBy = FakeResults.createFakeGroupByResult('@foo', 'a', 10);
+            results.groupByResults[0] = singleGroupBy;
+            resolve(results);
+          });
+          (<jasmine.Spy>mockEndpoint.search).and.returnValue(searchPromise);
+
+          facetSearchParams = new FacetSearchParameters(mockFacet);
+          facetSearchParams.nbResults = 10;
+        });
+
+        afterEach(() => {
+          facetSearchParams = null;
+          searchPromise = null;
+        });
+
+        it('should resolve with allowed values only', done => {
+          mockFacet.options.allowedValues = ['a0', 'a5', 'a8'];
+          facetQueryController.search(facetSearchParams).then(values => {
```

Comment:
You could use a helper method to generate your matcher arrays. It will be easier to read.